include/zephyr/net/coap_client.h

```diff
@@ -137,6 +137,16 @@ int coap_client_init(struct coap_client *client, const char *info);
 int coap_client_req(struct coap_client *client, int sock, const struct sockaddr *addr,
 		    struct coap_client_request *req, struct coap_transmission_parameters *params);
 
+/**
+ * @brief Cancel CoAP Requests
+ *
+ * When network connectivity is lost, terminate all pending requests.
```

Comment:
Cancellation should be possible even without any network errors.

For example CoAP block-wise transfer, we might see during the transfer that the FOTA image is going to be rejected. So no need to download rest of the blocks.